### PR TITLE
Fix for #1456 - ignore common authority override

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Instance/Authority.cs [hunk 1]
```diff
@@ -36,22 +27,65 @@ protected Authority(AuthorityInfo authorityInfo)
...
         #region Builders
-        public static Authority CreateAuthorityWithOverride(AuthorityInfo requestAuthorityInfo, AuthorityInfo configAuthorityInfo)
+
+        /// <summary>
+        /// Figures out the authority based on the authority from the config and the authority from the request, 
+        /// and optionally the homeAccountTenantId, which has an impact on AcquireTokenSilent
+        /// 
+        /// The algorithm is: 
+        /// 
+        /// 1. If there is no request authority (i.e. no authority override), use the config authority.
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Instance/Authority.cs]: >request [](start = 30, length = 7)

nit: the meaning of "request" authority is not very clear here, esp from a customer perspective. is it the authority used in the request?

File: tests/devapps/NetCoreTestApp/Program.cs [hunk 2]
```diff
@@ -107,88 +107,89 @@ 1. IWA
...
-                        return;
-                    default:
-                        break;
+                                BrowserRedirectSuccess = new Uri("https://www.bing.com?q=why+is+42+the+meaning+of+life")
+                            };
+
+                            var cts = new CancellationTokenSource();
+                            authTask = pca.AcquireTokenInteractive(s_scopes)
+                                .WithSystemWebViewOptions(options)
+                                 .WithExtraQueryParameters("dc=PROD-WST-CORP")
```
Comment (jennyf19) [inline tests/devapps/NetCoreTestApp/Program.cs]: do you want to leave this in?

File: tests/Microsoft.Identity.Test.Unit.net45/CacheTests/UnifiedCacheTests.cs [hunk 3]
```diff
@@ -77,6 +80,7 @@ public void UnifiedCache_MsalStoresToAndReadRtFromAdalCache()
                             TestConstants.s_scope.ToArray())
                     });
 
+
                 // Using RT from Adal cache for silent call
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit.net45/CacheTests/UnifiedCacheTests.cs]: nit: remove extra line?

File: tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/PublicClientApplicationTests.cs [hunk 4]
```diff
@@ -1200,6 +1200,8 @@ public async Task MultiTenantWithAuthorityOverrideAsync()
             }
         }
 
+
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/PublicClientApplicationTests.cs]: nit: remove extra lines?

File: src/client/Microsoft.Identity.Client/Instance/Authority.cs [hunk 5]
```diff
@@ -36,22 +27,65 @@ protected Authority(AuthorityInfo authorityInfo)
...
         {
-            switch (requestAuthorityInfo.AuthorityType)
+            if (configAuthorityInfo == null)
+            {
+                throw new ArgumentNullException(nameof(configAuthorityInfo));
+            }
+
+            switch (configAuthorityInfo.AuthorityType)
             {
+                // ADFS and B2C are tenantless, no need to consider tenant
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Instance/Authority.cs]: >B2C are tenantless [](start = 28, length = 18)

What you mean by tenantless here? B2C has a tenant.

File: src/client/Microsoft.Identity.Client/Instance/Authority.cs (hunk 1 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Instance/Authority.cs]: This is not a public comment, the class is internal. Request Authority is the same as "authority override"  - it is the authority that is specified in the AcquireToken* builder, e.g. 

` AcquireTokenSilent(...).WithAuthority(...)`

THis is different from the config authority, which is the one specified on the client object: 

`var pca = PublicClientApplicationBuilder.Create("clientid").WithAuthority(...)`

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Instance/Authority.cs]: Happy to rephrase in a way that makes more sense, please advise.

File: src/client/Microsoft.Identity.Client/Instance/Authority.cs (hunk 5 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Instance/Authority.cs]: From MSAL perspective, I don't think it does. If you use .WithTentantId() or overloads of WithAuthority() that take a tenant, I don't think it does anything to a B2C authority. 

Tenantless means "common", "organizations" or "consumers". Not sure how else to call it ... "commonish" ? :)

Comment (jennyf19) [reply src/client/Microsoft.Identity.Client/Instance/Authority.cs]: idk...but definitely don't say "commonish"

Comment (jmprieur) [reply src/client/Microsoft.Identity.Client/Instance/Authority.cs]: not-tenanted authorties?

**Pull request merged**